runtime/near-evm-runner/src/runner.rs

```diff
@@ -249,6 +304,7 @@ impl<'a> EvmContext<'a> {
             return Err(VMLogicError::EvmError(EvmError::InvalidNonce));
         }
         self.add_balance(&meta_call_args.sender, U256::from(self.attached_deposit))?;
+        // TODO: this is wrong?!
```

Comment:
what exactly is wrong?

Reply:
because the attached deposit coming from the relayer. Instead value should be attached by the meta call itself, charging the user that signed meta tx.
I think we are missing the `value` field in the meta call and should not allow to attach funds by relayers

Reply:
parse_meta_call returns fee value as fee_amount now:
```
    Ok(MetaCallArgs { sender, nonce, fee_amount, fee_address, contract_address, input })

```

Reply:
That's not the `value`.
`Value` is how much signer attached of base token with the call.
This considers attached deposit coming from relayer, which is incorrect.
We need to add another field to the meta call - `value` (or can call it attached_deposit or whatever) that will actually specify how much to deduct from given user (if they have enough balance).
See the implementation of raw_call for example of such deduction.

Reply:
Got it, some subword of `attached_(base_token)_deposit_value` :D